lib/lazy_doc/provider_behavior.ex

```diff
@@ -0,0 +1,66 @@
+defmodule LazyDoc.ProviderBehavior do
```

Comment:
I would call only `LazyDoc.Provider` (no behavior).

Although this module provides the callback contract (behaviour), it also contains the function `request_prompt/3`, which is being used to call the provider by the callback module. So it's doing a little bit more than just a behaviour definition

Reply:
Nice , I will change it tomorrow.

Reply:
[rename behavior](https://github.com/pxp9/lazy_doc/pull/3/commits/2cb690b556f6e0cda24b3a9d3499e95402d5b239)